packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -24,8 +130,112 @@ class DeepLinksScreen extends Screen {
 
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('TODO: build deep link validation tool'),
+    return const DeepLinkPage();
+  }
+}
+
+class DeepLinkPage extends StatefulWidget {
+  const DeepLinkPage({super.key});
+
+  @override
+  State<DeepLinkPage> createState() => _DeepLinkPageState();
+}
+
+class _DeepLinkPageState extends State<DeepLinkPage> {
+  String responseBody = 'empty response';
+  int? selectedRowIndex;
+  String searchContent = '';
+  bool bundleByDomain = false;
+
+  void setBundleByDomain(bool shouldBundleByDomain) {
+    setState(() {
+      bundleByDomain = shouldBundleByDomain;
+    });
+  }
+
+  void setResponse(String response) {
+    setState(() {
+      responseBody = response;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      children: [
+        Row(
+          children: [
+            Expanded(child: buildTitle('All deep links', context)),
+            InkWell(
+              onTap: () {
+                setBundleByDomain(!bundleByDomain);
+              },
+              child: const Text('Bundle by domain'),
+            ),
+            const SizedBox(width: 10),
+            SearchBar(
+              leading: const Icon(Icons.search),
+              onChanged: (value) {
+                setState(() {
+                  searchContent = value.toLowerCase();
+                });
+              },
+              constraints: BoxConstraints.tight(const Size(200, 40)),
+            ),
+          ],
+        ),
+        const SizedBox(height: 10),
+        buildDataTable(context),
+      ],
     );
   }
+
+  Widget buildDataTable(BuildContext context) {
```

Comment:
make this a stateless widget class instead of having a method that returns a widget

Reply:
Updated it to use 

ValueListenableBuilder<List<LinkData>>(
      valueListenable: controller.linkDatasNotifier,

so  it is not a  a stateless widget  now.

Reply:
While `ValueListenableBuilder` is not a stateless widget, a helper widget would still be stateless. In general, we never want to return a widget from a method. Always prefer a helper widget instead. 

```
class _DataTable extends StatelessWidget {
  ...
  
  Widget build(BuildContext context) {
    return ...;
  }
}
```

Reply:
done